release/src/Main.hs

```diff
@@ -44,6 +44,8 @@ main = do
                       $logInfo ("Writing slack release message to "# T.pack fp)
                       liftIO (BS.writeFile fp (T.encodeUtf8 (slackReleaseMessage os (renderVersion sdkVersion))))
                   releaseToBintray upload releaseDir artifacts
+                  $logInfo "##vso[task.setvariable variable=has_released]true"
```

Comment:
It would be nice to add a comment that explains this/links to the docs. I had to google this to understand how it works :smile:

I would also recommend that you use `putStrLn` instead of `$logInfo`. `$logInfo` might mess around with formatting and I am not sure how sensitive Azure is in that regard.

Reply:
`putStrLn` was my first attempts, but I could not get it to typecheck. Any suggestion welcome; error message is: 
```
release/src/Main.hs:47:19: error:
    • Couldn't match type ‘IO’ with ‘LoggingT IO’
      Expected type: LoggingT IO ()
        Actual type: IO ()
    • In a stmt of a 'do' block:
        putStrLn "##vso[task.setvariable variable=has_released]true"
      In the expression:
        do ((((monadLoggerLog
                 (((((Loc "release/src/Main.hs") "main") "Main") (42, 19))
                    (42, 27)))
                (T.pack ""))
               LevelInfo
               . (id :: T.Text -> T.Text)))
             "Make release"
           for_ optsSlackReleaseMessageFile $ \ fp -> do ...
           releaseToBintray upload releaseDir artifacts
           putStrLn "##vso[task.setvariable variable=has_released]true"
           ....
```

Reply:
Try `liftIO $ putStrLn "…"`. `liftIO` allows you to lift an action of type `IO a` into anything that is an instance of `MonadIO` and `LoggingT IO` is an instance of `MonadIO`.

Reply:
Re: the comment I put an explanation in the commit message. Is that not a location people look at here?

Reply:
I generally prefer also having an inline source code comment instead of purely relying on information in commit messages.